x-pack/plugins/ml/public/data_frame/common/kibana_context.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { StaticIndexPattern } from 'ui/index_patterns';
+
+interface KibanaContextValue {
+  currentIndexPattern: StaticIndexPattern;
+}
+
+interface KibanaExtendedContextValue extends KibanaContextValue {
+  indexPatterns: any;
+  kibanaConfig: any;
```

Comment:
could these be optional properties of `KibanaContextValue`?

Reply:
You're right that would work! I had another look at the code and it turned out the two different types are not necessary at all. I was wrong assuming the type with `currentIndexPattern` only was used for the jobs list but that's not the case. I fixed that in 9a3faed.